mittab/apps/tab/middleware.py

```diff
@@ -1,15 +1,29 @@
+import re
+
 from django.contrib.auth.views import login
 from django.http import HttpResponseRedirect
-from django.core.cache import cache
 
-login_white_list = ('/accounts/login/', '/static/css/stylesheet.css', '/static/images/title_banner.png', '/pairings/pairinglist/')
+from mittab.apps.tab.models import TabSettings
+
+login_white_list = ('/accounts/login/', '/static/css/stylesheet.css',
+        '/static/images/title_banner.png', '/pairings/pairinglist/', '/stat')
+
+e_ballot_regex = re.compile("/e_ballots/\S+")
 #backend_white_list = ('/static/css/stylesheet.css', '/enter_tab/', '/static/images/title_banner.png')
 
 class Login:
     "This middleware requires a login for every view"
     def process_request(self, request):
-        if request.path not in login_white_list and request.user.is_anonymous():
+        whitelisted = (request.path in login_white_list)
+        whitelisted = whitelisted or self._is_eballot_url(request.path)
```

Comment:
I think this needs to just be self._is_eballot_url(request)

and then later instead of request == ... do request.path ==